src/dataModels/JSON.js

```diff
@@ -832,7 +831,12 @@ function reselectGridRowsBackedBySelectedDataRows() {
 
 function getDataSourceName(name) {
     name = analytics[name].prototype.$$CLASS_NAME || name;
-    return name.replace(/^Data(Source|Node)/, '').toLowerCase() || 'source';
+    if (/^DataSource\w+Filter$/.test(name)) {
+        name = 'filter';
```

Comment:
_Explanation:_ Any data source with the word "Filter" in it becomes `sources.filter`